hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/ColumnStatsIndexSupport.scala

```diff
@@ -209,11 +209,11 @@ class ColumnStatsIndexSupport(spark: SparkSession,
     // NOTE: We're sorting the columns to make sure final index schema matches layout
     //       of the transposed table
     val sortedTargetColumnsSet = TreeSet(queryColumns:_*)
-    val sortedTargetColumns = sortedTargetColumnsSet.toSeq
 
     // NOTE: This is a trick to avoid pulling all of [[ColumnStatsIndexSupport]] object into the lambdas'
     //       closures below
     val indexedColumns = this.indexedColumns
+    val indexedTargetColumns = sortedTargetColumnsSet.filter(indexedColumns.contains(_)).toSeq
```

Comment:
Let's de-duplicate filtering and tie it up w/ index schema composition:

 - Let's make `composeIndexSchema` return (schema, targetIndexedColumns)
 - Let's move schema composition up here

Reply:
Thanks,I've re-modified.